composer/callbacks/free_outputs.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Free train metrics ."""
+
+import torch
+
+from composer.core import Callback, State
+from composer.loggers import Logger
+
+
+class FreeOutputs(Callback):
+    """Free train metrics on AFTER_LOSS to reduce peak memory usage if not using train metrics."""
```

Comment:
QQ: where is the "not using train metrics implemented"?

Reply:
User is free to pass them in if they want. https://github.com/mosaicml/llm-foundry/pull/642 here is a foundry PR to add that option in our stack